test/helpers/kubectl.go

```diff
@@ -800,46 +800,50 @@ func (kub *Kubectl) CiliumEndpointsListByLabel(pod, label string) (EndpointMap,
 	return result, nil
 }
 
-// CiliumEndpointWait waits until all endpoints managed by the specified Cilium
-// pod are ready. Returns false if the command to retrieve the state of the
-// endpoints times out.
-func (kub *Kubectl) CiliumEndpointWait(pod string) bool {
+// CiliumEndpointWaitReady waits until all endpoints managed by all Cilium pods
+// pod are ready. Returns an error if cannot get the cilium pods, or the
+// endpoint ready times out.
+func (kub *Kubectl) CiliumEndpointWaitReady() error {
+	ciliumPods, err := kub.GetCiliumPods(KubeSystemNamespace)
+	if err != nil {
+		kub.logger.WithError(err).Error("cannot get Cilium pods")
+		return err
+	}
 
 	body := func() bool {
-		status, err := kub.CiliumEndpointsList(pod).Filter("{range [*]}{.status.state},{.status.identity.id} {end}")
-		if err != nil {
-			return false
-		}
-
-		var valid, invalid int
-		for _, endpoint := range strings.Split(strings.TrimRight(status.String(), " "), " ") {
-			fields := strings.Split(endpoint, ",")
-			state := fields[0]
-			secID := fields[1]
-			// Consider an endpoint with reserved identity 5 (reserved:init) as not ready.
-			if state != "ready" || secID == "5" {
-				invalid++
-			} else {
-				valid++
+		for _, pod := range ciliumPods {
+			status, err := kub.CiliumEndpointsList(pod).Filter(`{range [*]}{.status.state}{"="}{.status.identity.id}{"\n"}{end}`)
+			if err != nil {
+				kub.logger.WithError(err).Errorf("cannot get endpoints state on pod '%s'", pod)
+				return false
 			}
-		}
-		if invalid == 0 {
-			return true
-		}
 
-		kub.logger.WithFields(logrus.Fields{
-			"pod":     pod,
-			"valid":   valid,
-			"invalid": invalid,
-		}).Info("Waiting for cilium endpoints")
-		return false
-	}
+			total := 0
+			invalid := 0
+			for _, line := range strings.Split(status.String(), "\n") {
+				total++
+				vals := strings.Split(line, "=")
+				if len(vals) != 2 {
```

Comment:
Add a comment in the code which explains what the desired output looks like (i.e., add an example).